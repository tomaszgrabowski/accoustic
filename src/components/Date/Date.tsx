import dayjs from 'dayjs';
import React from 'react';

interface IProps {
    date: string;
    format: string;
}

const Date = ( props: IProps ) => {
    const isValid = dayjs( props.date ).isValid();
    return (
        isValid ?
            <small>
                { dayjs( props.date ).format( props.format ) }
            </small>
            : <small>
                Date not valid
            </small>
    );
    
};

export default Date;
