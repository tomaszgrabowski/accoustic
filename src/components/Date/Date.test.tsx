import dayjs from 'dayjs';
import { shallow } from 'enzyme';
import * as faker from 'faker';
import React from 'react';
import renderer from 'react-test-renderer';
import Date from './Date';

describe( 'Date', () => {
    test( 'given date when render then show date', () => {
        const date = faker.date.recent().toISOString();
        const formattedDate = dayjs( date ).format( 'DD-mm-YYYY HH:mm' );
        const wrapper = shallow( <Date date={ date } format={ 'DD-mm-YYYY HH:mm' }/> );
        expect( wrapper.find( 'small' ).text() ).toEqual( formattedDate );
    } );
    
    test( 'given invalid date when render then show notification', () => {
        const date = 'incorrect date';
        const wrapper = shallow( <Date date={ date } format={ 'DD-mm-YYYY HH:mm' }/> );
        expect( wrapper.find( 'small' ).text() ).toEqual( 'Date not valid' );
    } );
    test( 'given Date component when render then renders properly', () => {
        const wrapper = renderer.create( <Date date={ '2012-01-01T23:23:23Z' } format={ 'DD-mm-YYYY HH:mm' }/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
    
} );
