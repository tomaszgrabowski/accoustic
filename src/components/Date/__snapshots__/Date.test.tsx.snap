// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date given Date component when render then renders properly 1`] = `
<small>
  02-23-2012 00:23
</small>
`;
